code/game/gamemodes/miniantags/pulsedemon/pulsedemon_interactions.dm

```diff
@@ -0,0 +1,272 @@
+	user.forceMove(src)
+	if(user.check_valid_recharger(src))
+		if(!user.pb_helper.start(user, src, user.hijack_time, TRUE, PD_HIJACK_CB(user, check_valid_recharger), PD_HIJACK_CB(user, finish_hijack_recharger), PD_HIJACK_FAIL(user)))
+			to_chat(user, "<span class='warning'>You are already performing an action!</span>")
+		else
+			user.do_hijack_notice(charging)
+	else
+		to_chat(user, "<span class='warning'>There is no weapon charging. Click again to retry.</span>")
+
+/mob/living/simple_animal/pulse_demon/proc/check_valid_recharger(obj/machinery/recharger/R)
+	return R.charging
+
+/mob/living/simple_animal/pulse_demon/proc/finish_hijack_recharger(obj/machinery/recharger/R)
+	to_chat(src, "<span class='notice'>You are now inside [R.charging]. Click on a hijacked APC to return.</span>")
+	forceMove(R.charging)
+	current_weapon = R.charging
+
+/obj/machinery/cell_charger/attack_pulsedemon(mob/living/simple_animal/pulse_demon/user)
+	user.forceMove(src)
+	if(user.check_valid_cell_charger(src))
+		if(charging.rigged)
+			user.finish_hijack_cell_charger(src)
+			return
+		if(!user.pb_helper.start(user, src, user.hijack_time, TRUE, PD_HIJACK_CB(user, check_valid_cell_charger), PD_HIJACK_CB(user, finish_hijack_cell_charger), PD_HIJACK_FAIL(user)))
+			to_chat(user, "<span class='warning'>You are already performing an action!</span>")
+		else
+			user.do_hijack_notice(charging)
+	else
+		to_chat(user, "<span class='warning'>There is no cell charging. Click again to retry.</span>")
+
+/mob/living/simple_animal/pulse_demon/proc/check_valid_cell_charger(obj/machinery/cell_charger/C)
+	return C.charging
+
+/mob/living/simple_animal/pulse_demon/proc/finish_hijack_cell_charger(obj/machinery/cell_charger/C)
+	to_chat(src, "<span class='notice'>You are now inside [C.charging]. Click on a hijacked APC to return.</span>")
+	forceMove(C.charging)
+	C.charging.rigged = TRUE
+
+/obj/machinery/recharge_station/attack_pulsedemon(mob/living/simple_animal/pulse_demon/user)
+	user.forceMove(src)
+	if(user.check_valid_recharge_station(src))
+		var/mob/living/silicon/robot/R = occupant
+		if(R in user.hijacked_robots)
+			user.finish_hijack_recharge_station(src)
+			return
+		if(!user.pb_helper.start(user, src, user.hijack_time, TRUE, PD_HIJACK_CB(user, check_valid_recharge_station), PD_HIJACK_CB(user, finish_hijack_recharge_station), PD_HIJACK_FAIL(user)))
+			to_chat(user, "<span class='warning'>You are already performing an action!</span>")
+		else
+			user.do_hijack_notice(occupant)
+			to_chat(R, "<span class='danger'>ALERT: ELECTRICAL MALEVOLENCE DETECTED, TARGETING SYSTEMS HIJACK IN PROGRESS</span>")
+	else
+		to_chat(user, "<span class='warning'>There is no silicon-based occupant inside. Click again to retry.</span>")
+
+/mob/living/simple_animal/pulse_demon/proc/check_valid_recharge_station(obj/machinery/recharge_station/R)
+	return R.occupant && istype(R.occupant, /mob/living/silicon/robot)
```

Comment:
```suggestion
	return isrobot(R.occupant)
```
istype will automatically null check `R.occupant` and we have a define for this type check aswell